### ISSUE TITLE
Turn exercise 006 into practice

### ISSUE THREAD

**Issue by cuihtlauac** [ocaml/ocaml.org#2172]
Extract the OCaml code from the markdown file corresponding to this exercise. The file is: `data/exercise/006_NAME.md` (where `006` is the number from the title of this issue and `NAME` is the rest of the file name, for instance `data/exercise/001_tail.md` is the file which corresponds to exercise `001`). Use the code in this file to create a folder `practice/006` (using this same number).

Populate the `006` folder with (almost) the same files as in folders `practice/001` and `practice/002` (they are examples).

Three files are different in each folder:
1. `practice/006/ex.ml`: it contains the function's signature, which needs to be implemented by the learner; 
1. `practice/006/work/impl.ml`: it includes the template meant to be filed by the learner;
1. `practice/006/answer/impl.ml`: consists of the solution from the original markdown file.

The examples included in the original markdown file must be turned into test cases in the `practice/006/ex.ml` file.

Folder structure to create:
```
practice/006/
├── answer
│   ├── impl.ml
│   └── test
│       ├── dune
│       └── run.ml
├── dune
├── dune-project
├── ex.ml
└── work
    ├── impl.ml
    └── test
        ├── dune

**Comment by Ozyugoo** [comment]
@cuihtlauac can I work on this if the 004 is what you expect me to do?

**Comment by cuihtlauac** [comment]
@Ozyugoo: complete 004 before picking another one.

**Comment by Ozyugoo** [comment]
> @Ozyugoo: complete 004 before picking another one.

Alright Thanks.

**Comment by maha-sachin** [comment]
Hi @cuihtlauac @SaySayo , I’d like to work on this.

**Comment by divyankachaudhari** [comment]
Hi, @cuihtlauac I'd like to work on this, I've completed a draft for another issue that I'd been working on. Since @maha-sachin seems to be working on #2180, I'll start my work on this issue. Thank you :)

**Comment by cuihtlauac** [comment]
Closed by #2217
